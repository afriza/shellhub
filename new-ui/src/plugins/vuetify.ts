// Styles
import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";
import "../styles/_variables.scss";

// Vuetify
import { createVuetify } from "vuetify";

const light = {
  dark: false,
  colors: {
    primary: "#667acc",
    secondary: "#c4c4c4",
    background: "#FFFFFF",
    tabs: "#FFFFFF",
    foreground: "#FFFFFF",
    paymentForm: "#FFFFFF",
  },
};

const dark = {
  dark: true,
  colors: {
    primary: "#667acc",
    secondary: "#1E2127",
    background: "#18191B",
    tabs: "#1E1E1E",
    foreground: "#1E1E1E",
    paymentForm: "#E0E0E0",
  },
};

export default createVuetify({
  theme: {
    defaultTheme: "dark",
    themes: {
      dark,
      light,
    },
  },
});
