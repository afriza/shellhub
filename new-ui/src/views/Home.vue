<template>
  <h1>Teste !</h1>
  <v-btn>Hello</v-btn>
  <router-link to="/teste">Teste</router-link>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "Home",
  setup() {

    return {
      //
    };
  },
});
</script>
