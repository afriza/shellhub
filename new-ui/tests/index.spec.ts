import { createVuetify } from "vuetify";
import { mount } from "@vue/test-utils";
import { beforeEach, describe, expect, it } from "vitest";
import Home from "../src/views/Home.vue";

describe("Home", () => {
  let wrapper;

  beforeEach(() => {
    const vuetify = createVuetify();

    wrapper = mount(Home, {
      global: {
        plugins: [vuetify],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper).toBeTruthy();
  });
  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
