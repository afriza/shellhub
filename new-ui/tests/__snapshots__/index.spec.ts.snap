// Vitest Snapshot v1

exports[`Home > Renders the component 1`] = `
"<h1>Teste !</h1>
<button type=\\"button\\" class=\\"v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-elevated\\">
  <div class=\\"v-btn__overlay\\"></div>
  <div class=\\"v-btn__underlay\\"></div>
  <!---->
  <div class=\\"v-btn__content\\" data-no-activator=\\"\\">Hello</div>
  <!---->
  <!---->
</button>
<router-link to=\\"/teste\\">Teste</router-link>"
`;
